### PR TITLE
commit

### PR DESCRIPTION
**Pull request by Pankaj-Kumar110**
User history + landing page fix

### CODE REVIEW

File: src/_mock/_userHistory.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+][index];
+
+
+export interface History {
+  id: string;
+  source: string;
+  destination: string;
+  rating: number;
+  travelMode: string;
+}
```
Comment (Akshay-Tomar-1135) [inline src/_mock/_userHistory.ts]: move this interface to ```src/interface``` folder in a new file named as ```userHistory.ts```.
Declare all interfaces or types related to 'User History' functionality there.

File: src/components/Header.tsx [hunk 2]
```diff
@@ -71,21 +72,7 @@ export default function Header() {
...
-              </li>
-              <li>
-                <a href="#">Feedback</a>
-              </li>
-              <li>
-                <a href="#">Help</a>
-              </li>
+              {renderNavLinks()}
               <Button text="Get Started" onButtonClick={() => router.push('/auth')} />
             </ul>
```
Comment (Akshay-Tomar-1135) [inline src/components/Header.tsx]: this should be writtern as the following in line 67
```
<Button text="Get Started" onButtonClick={() => setOpenModal(true)} />
```

File: src/components/Header.tsx
Comment (Akshay-Tomar-1135) [inline src/components/Header.tsx:1]: modify the following file like this
```
import { useState, useEffect } from 'react';
import Logo from '/assets/images/others/main-logo.png';
import { FaBars } from 'react-icons/fa';
import Box from '@mui/material/Box';
import { useRouter } from 'src/routes/hooks';
import { useAppDispatch } from 'src/store/reduxHooks';
import { setUserType } from 'src/store/features/userConfig/userConfigSlice';
import { UserTypes } from 'src/_mock';
import UserTypeModal from './UserTypeModal';
import Button from './Button';

export default function Header() {
  const [localUserType, setLocalUserType] = useState<UserTypes | null>(null);
  const [openModal, setOpenModal] = useState<boolean>(false);
  const router = useRouter();
  const [isMenuOpen, setIsMenuOpen] = useState(false); // State to track menu visibility

  const toggleMenu = () => {
    setIsMenuOpen(!isMenuOpen); // Toggle menu visibility
  };

  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(setUserType(localUserType));
  }, [dispatch, localUserType]);

  useEffect(() => {
    console.log('userType in header ---', localUserType);
    if (!localUserType) return;
    router.push('/auth');
  }, [localUserType, router]);

  const navLinks = [
    { name: 'About', href: '#about' },
    { name: 'Features', href: '#features' },
    { name: 'Customers', href: '#customers' },
    { name: 'FAQ', href: '#faq' },
    { name: 'Feedback', href: '#feedback' },
  ];

  const renderNavLinks = () =>
    navLinks.map((link) => (
      <li key={link.name}>
        <a href={link.href}>{link.name}</a>
      </li>
    ));

  const menuList = (
    <>
      {renderNavLinks}
      <Button text="Get Started" onButtonClick={() => setOpenModal(true)} />
    </>
  );

  return (
    <>
      <UserTypeModal
        openModal={openModal}
        handleCloseModal={() => setOpenModal(false)}
        setUserType={setLocalUserType}
      />

      <header className="flex justify-between items-center px-6 py-4 backdrop-blur-md sticky top-0 z-20 bg-gradient-to-r from-[#E0E7FD] to-[#FDFEFF] shadow-md">
        <Box component="img" src={Logo} alt="Logo" className="cursor-pointer" />

        {/* Menu Icon for Mobile */}
        <FaBars className="block md:hidden cursor-pointer" onClick={toggleMenu} />

        {/* Desktop Navigation */}
        <nav className="hidden md:block">
          <ul className="flex gap-6 items-center">{menuList}</ul>
        </nav>

        {/* Mobile Navigation */}
        {isMenuOpen && (
          <nav className="md:hidden absolute top-full left-0 w-full bg-gradient-to-r from-[#E0E7FD] to-[#FDFEFF] shadow-md">
            <ul className="flex flex-col gap-4 p-6">{menuList}</ul>
          </nav>
        )}
      </header>
    </>
  );
}

```

this way we will have one common variable (menuIList) to edit the menu list in every device width

File: src/layouts/dashboard/nav.tsx [hunk 3]
```diff
@@ -119,7 +119,7 @@ export function NavContent({ data, slots, workspaces, sx }: NavContentProps) {
 
   return (
     <>
-      <Logo />
+      <img src={Logo} alt="Logo" style={{height:"50px",width:"50px"}} />
```
Comment (Akshay-Tomar-1135) [inline src/layouts/dashboard/nav.tsx:124]: use tailwind CSS instead of inline syling

File: src/layouts/simple/layout.tsx [hunk 4]
```diff
@@ -43,7 +43,11 @@ export function SimpleLayout({ sx, children, header, content }: SimpleLayoutProp
                 This is an info Alert.
               </Alert>
             ),
-            leftArea: <Logo />,
+            leftArea: (<img
+              src={Logo}
+              alt="Logo"
+              style={{ height: '50px',width:"50px"}}
+            />),
```
Comment (Akshay-Tomar-1135) [inline src/layouts/simple/layout.tsx]: use tailwind CSS

**Review by Akshay-Tomar-1135 — CHANGES_REQUESTED** [review]
develop branch has been updated
* take fresh pull from develop and merge develop's changes to your branch
* resolve conflicts
* resolve PR comments
* then push you changes
* whoever between you and @adityaS-99 resolves his PR comments first, his PR gets merged to develop first, then other one has to merge develop again to his branch and resolve conflict and then push changes, after that his PR will get merged, if having all PR comments resolved

**Review by Pankaj-Kumar110 — COMMENTED** [review]
Corrected all things which was requested like 
1. improved user History styling
2. added interface in separate file
3. fixed nav links

File: src/_mock/_userHistory.ts (hunk 1 above)
Comment (Akshay-Tomar-1135) [reply src/_mock/_userHistory.ts]: still this interface is not removed from here

File: src/app.tsx [hunk 5]
```diff
@@ -5,8 +5,6 @@ import { useScrollToTop } from 'src/hooks/use-scroll-to-top';
 import { ThemeProvider } from 'src/theme/theme-provider';
 import { Iconify } from 'src/components/iconify';
 import { Provider } from 'react-redux';
-import { SnackbarProvider } from 'notistack';
-import { ToastProvider } from './components/snackBar/ToastContext';
 import store from './store/store';
```
Comment (Akshay-Tomar-1135) [inline src/app.tsx:10]: why removed these changes, these were important

File: src/app.tsx [hunk 6]
```diff
@@ -37,12 +35,7 @@ export default function App() {
...
     <Provider store={store}>
       <ThemeProvider>
-        <SnackbarProvider maxSnack={3}>
-          <ToastProvider>
-            <Router />
-            {githubButton}
-          </ToastProvider>
-        </SnackbarProvider>
+        <Router />
       </ThemeProvider>
```
Comment (Akshay-Tomar-1135) [inline src/app.tsx:45]: only remove 
```
{githubButton}
```

File: src/layouts/auth/layout.tsx [hunk 7]
```diff
@@ -49,7 +49,11 @@ export function AuthLayout({ sx, children, header }: AuthLayoutProps) {
                 This is an info Alert.
               </Alert>
             ),
-            leftArea: <Logo />,
+            leftArea: (<img
+              src={Logo}
+              alt="Logo"
+              style={{ height: '50px',width:"50px"}}
+            />),
```
Comment (Akshay-Tomar-1135) [inline src/layouts/auth/layout.tsx]: use tailwind css

File: src/layouts/simple/layout.tsx (hunk 4 above)
Comment (Akshay-Tomar-1135) [reply src/layouts/simple/layout.tsx]: if you are giving custome value to height and width, then give it 'rem' not in 'px'.
may be that's the best practice.

File: src/layouts/auth/layout.tsx [hunk 8]
```diff
@@ -49,7 +49,7 @@ export function AuthLayout({ sx, children, header }: AuthLayoutProps) {
                 This is an info Alert.
               </Alert>
             ),
-            leftArea: <Logo />,
+            leftArea: <img src={Logo} alt="Logo" className="h-[50] w-[50]" />,
```
Comment (Akshay-Tomar-1135) [inline src/layouts/auth/layout.tsx]: what is 50 put some unit like ```rem```

**Review by Akshay-Tomar-1135 — CHANGES_REQUESTED** [review]
fix the one comment

**Review by Akshay-Tomar-1135 — APPROVED** [review]
all good

**Pull request merged**